flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testutils/source/reader/SourceReaderTestBase.java

```diff
@@ -49,9 +50,17 @@ Licensed to the Apache Software Foundation (ASF) under one
  */
 public abstract class SourceReaderTestBase<SplitT extends SourceSplit> extends TestLogger {
 
-    protected static final int NUM_SPLITS = 10;
+    protected static int numSplits;
```

Comment:
Hmm, if this variable is static and more than 2 test cases inheriting the `SourceReaderTestBase` are running concurrently, they may interfere with each other. We probably want to make this variable non-static.

Maybe we can do the following:
1. Make `numSplits` and `totalNumRecords` private final.
2. Add a new method `int getNumSplits()`;
3. In the default constructor of `SourceReaderTestBase`, set the  `numSplits` to `getNumSplits()` and compute the `totalNumRecords` accordingly.

The concrete test classes can then change the numSplits by overriding the method of `getNumSpltis()`.

Reply:
Thanks for the comments. I have done the 2 and 3 in commit 56843f8 . For 1, I left `numSplits` and `totalNumRecords` protected final because this can help the direct accesses to these variables in the Kafka test class, and the `final` can still avoid further changes of these variables. If you still have concerns on this, I can still change, thanks.